core/templates/pages/collection-editor-page/navbar/collection-editor-navbar.directive.spec.ts

```diff
@@ -0,0 +1,549 @@
+
+    expect(ctrl.collection).toBe(sampleCollection);
+  });
+
+  it('should unpublish collection successfully ' +
+    'when unpublish button is clicked', fakeAsync(function() {
+    ctrl.collection = sampleCollection;
+    ctrl.collectionRights = new CollectionRights({
+      owner_names: [],
+      collection_id: 1,
+      can_edit: true,
+      can_unpublish: true,
+      is_private: false
+    });
+    spyOn(collectionRightsBackendApiService, 'setCollectionPrivateAsync')
+      .and.returnValue(Promise.resolve(null));
+    let collectionRightsSpy = spyOn(
+      collectionEditorStateService, 'setCollectionRights')
+      .and.returnValue(null);
+
+    ctrl.unpublishCollection();
+    tick();
+
+    expect(collectionRightsSpy).toHaveBeenCalled();
+  }));
+
+  it('should not unpublish collection in case of ' +
+    'backend error', fakeAsync(function() {
+    ctrl.collection = sampleCollection;
+    ctrl.collectionRights = sampleCollectionRights;
+    spyOn(collectionRightsBackendApiService, 'setCollectionPrivateAsync')
+      .and.returnValue(Promise.reject(null));
+    let collectionRightsSpy = spyOn(
+      collectionEditorStateService, 'setCollectionRights')
+      .and.returnValue(null);
+
+    ctrl.unpublishCollection();
+    tick();
+
+    expect(collectionRightsSpy).not.toHaveBeenCalled();
+  }));
+
+  it('should open a modal to save changes if ' +
+    'additional meta data needed', fakeAsync(function() {
+    sampleCollection = Collection.create(
+      sampleCollectionBackendObject2);
+    ctrl.collection = sampleCollection;
+    ctrl.collectionRights = new CollectionRights({
+      owner_names: [],
+      collection_id: 1,
+      can_edit: true,
+      can_unpublish: true,
+      is_private: false
+    });
+
+    spyOn($uibModal, 'open').and.returnValue({
+      result: Promise.resolve(['metadata'])
+    });
+    spyOn(urlInterpolationService, 'interpolateUrl')
+      .and.returnValue('url');
+    let saveSpy = spyOn(
+      collectionEditorStateService, 'saveCollection')
+      .and.returnValue(null);
+
+    ctrl.publishCollection();
+    $rootScope.$apply();
+    tick();
+    $rootScope.$apply();
+
+    expect(saveSpy).toHaveBeenCalled();
+  }));
+
+  it('should publish collection successfully if no ' +
+    'additional meta data needed', fakeAsync(function() {
+    ctrl.collection = sampleCollection;
+    ctrl.collectionRights = new CollectionRights({
+      owner_names: [],
+      collection_id: 1,
+      can_edit: true,
+      can_unpublish: true,
+      is_private: false
+    });
+
+    spyOn(collectionRightsBackendApiService, 'setCollectionPublicAsync')
+      .and.returnValue(Promise.resolve(null));
+    let collectionRightsSpy = spyOn(
+      collectionEditorStateService, 'setCollectionRights')
+      .and.returnValue(null);
+
+    spyOn(urlInterpolationService, 'interpolateUrl')
+      .and.returnValue('url');
+
+    ctrl.publishCollection();
+    $rootScope.$apply();
+    tick();
+    $rootScope.$apply();
```

Comment:
Do we need to call `$rootScope` twice? If that is the case can you add a comment explaining why?

Reply:
done